### PR TITLE
Ballot test

### PR DESCRIPTION
**Pull request by Hazardu**
## Description
Added a test for ballot bit count functions from `nbl/builtin/glsl/workgroup/ballot.glsl` to example 48.ArithmeticUnitTest
Additionally, after merging I encountered problems with CMeshManipulator.cpp, the template parameter count changed. Should this change be undone or is this ok?

## TODO list:
- [x] `#include <nbl/builtin/glsl/subgroup/arithmetic_portability_impl.glsl>` is causing compilation errors, despite the file being where it should be.
- [x] last time, bitCount functions were all returning 0, I need to confirm what the correct, expected output should be. I previously understood it should either return 0 or 1, as the value to be counted was a boolean (which also take values of that range), and it should sum the results of previous invocations within the workgroup.

### CODE REVIEW

File: src/nbl/asset/CMeshManipulator.cpp [hunk 1]
```diff
@@ -1535,21 +1535,21 @@ bool CMeshManipulator::calcMaxQuantizationError(const SAttribTypeChoice& _srcTyp
 	return true;
 }
 
-core::smart_refctd_ptr<ICPUBuffer> IMeshManipulator::idxBufferFromTriangleStripsToTriangles(const void* _input, size_t _idxCount, E_INDEX_TYPE _idxType)
+core::smart_refctd_ptr<ICPUBuffer> IMeshManipulator::idxBufferFromTriangleStripsToTriangles(const void* _input, size_t _idxCount, E_INDEX_TYPE _idxType, E_INDEX_TYPE _outIndexType)
```
Comment (Crisspl) [inline src/nbl/asset/CMeshManipulator.cpp:1538]: what's going on? Why output param is pass-by-value and whatsmore its not even used??

Comment (devshgraphicsprogramming) [reply src/nbl/asset/CMeshManipulator.cpp:1538]: my code, we've went over it

**Comment by devshgraphicsprogramming** [comment]
> is causing compilation errors, despite the file being where it should be.

Well, no shit, because all the macros and fuctions there are prefixed with `irr` and not `nbl`

**Comment by devshgraphicsprogramming** [comment]
does it pass tests?

**Comment by devshgraphicsprogramming** [comment]
ok I need to fix the ballot for WG>32

**Pull request merged**